### ISSUE TITLE
unable to parse package name "libxgboost-0.6+218_omp_0.tar.bz2"

### ISSUE THREAD

**Issue by ChristopherRussell** [mamba-org/rattler#758]
Hello. I get an error when parsing repodata.json of a channel (internal channel at work). I already posted an issue to rattler-build here https://github.com/prefix-dev/rattler-build/issues/944 and @wolfv suggested I should post here instead. The error message was:

```
ERROR resolve_conda{group=default, platform=linux-64}: rattler_repodata_gateway::gateway: error=failed to parse repodata.json
  failed to parse repodata.json
    invalid filename at line 38856 column 38 
```

and the snippet below shows the entry starting at the 38856'th line (or 'line 38855' if you call the first line 'line 0'). I don't think the error refers to the previous line though as it's only a few characters long (a few spaces plus a close bracket). We also have noarch and win-64 sections in the channel, but the repodata.json files for those are shorter than 38856 lines so can't be the cause of the error.

```
    },
    "libxgboost-0.6+218_omp_0.tar.bz2": {     <---- Line 38856
      "build": "0",
      "build_number": 0,
      "depends": [],
      "extra": {
        "git": {
          "branch": "HEAD",
          "commit": "2715bae",
          "sha": "2715baef64c07546314977caf858d1cce1043b66"
        },
        "package": {
          "md5": "43b62a5ba9470354c6244353bfcad65d"
        },
        "system": {
          "cwd": "/some/path/to/a/dir",
          "node": "name.company.com",
          "os": "Linux",
          "prefix": "/some/path",
          "time": "2017-04-06 14:13:26.296584",
          "user": "some_user"
        },
        "versions": {
         "conda": "4.1.12",
          "conda-build": "1.21.7",
          "conda-tox": "0.6.5",
          "python": "3.5.2"
        }
      },
      "features": "omp",
      "md5": "477ec8a7faa700d27138e1cdf8fec967",
      "name": "libxgboost",
      "sha256": "4005a7034a294aa0d063aa5c4273f09e21cd04e1997e436f5bf0aada7e5c91da",
      "size": 912390,
      "subdir": "linux-64",
      "version": "0.6+218"
    },
```

**Comment by wolfv** [comment]
Yes, that filename has only a single `-`. We assume that a valid filename has _at least_ two dashes (e.g. `libxgboost-0.6+218-omp_0` would be valid).

You can also see that somethign went wrong because the `build` is just `"0"` while it probably is mean to be something like `_omp_0`.

Not sure if we can solve this on our end easily. We could be more lenient but its' really quite broken metadata wise.

**Comment by ChristopherRussell** [comment]
If there's some pieces like the build that are just broken then fair enough. I don't know how this particular entry came to exist. 

I think more specific logging would have helped me with diagnosing the problem or getting help. Like 'could not parse the entry for libxgboost-0.6+218-omp_0 on line X column Y' would have been slightly nicer than just line X column Y. And anything else you can give 😄

So I think your point about better logging was a good one.

**Comment by wolfv** [comment]
Yep - we already merged a PR for that (#222)

**Comment by baszalmstra** [comment]
Closing this since #757 was merged.
